mlprimitives/custom/timeseries_preprocessing.py

```diff
@@ -4,21 +4,109 @@
+
+    Returns:
+        ndarray:
+            Array of boolean values, with one boolean value for each index value (`True` if the
+            index value is contained in a given interval, otherwise `False`).
+    """
+    mask = list()
+
+    if intervals is None or len(intervals) == 0:
+        return np.zeros((len(index), ), dtype=bool)
+
+    for value in index:
+        for start, end in intervals:
+            if start <= value <= end:
+                mask.append(True)
+                break
+        else:
+            mask.append(False)
+
+    return np.array(mask)
+
+
+def rolling_window_sequences(X, index, window_size, target_size, step_size, target_column,
+                             drop=None, drop_windows=False):
+    """Create rolling window sequences out of time series data.
+
+    The function creates an array of input sequences and an array of target sequences by rolling
+    over the input sequence with a specified window.
+    Optionally, certain values can be dropped from the sequences.
+
+    Args:
+        X (ndarray):
+            N-dimensional sequence to iterate over.
+        index (ndarray):
+            Array containing the index values of X.
+        window_size (int):
+            Length of the input sequences.
+        target_size (int):
+            Length of the target sequences.
+        step_size (int):
+            Indicating the number of steps to move the window forward each round.
+        target_column (int):
+            Indicating which column of X is the target.
+        drop (ndarray or None or str or float or bool):
+            Optional. Array of boolean values indicating which values of X are invalid, or value
+            indicating which value should be dropped. If not given, `None` is used.
+        drop_windows (bool):
+            Optional. Indicates whether the dropping functionality should be enabled. If not
+            given, `False` is used.
+
+    Returns:
+        ndarray:
+            Array containing the input sequences.
+        ndarray:
+            Array containing the target sequences.
+        ndarray:
+            Array containing the first index value of each input sequence.
+        ndarray:
+            Array containing the first index value of each target sequence.
+    """
     out_X = list()
     out_y = list()
     X_index = list()
     y_index = list()
-
     target = X[:, target_column]
 
-    for start in range(len(X) - window_size - target_size + 1):
+    if drop_windows:
+        if hasattr(drop, '__len__') and (not isinstance(drop, str)):
+            if len(drop) != len(X):
+                raise Exception('Arrays `drop` and `X` must be of the same length.')
+        else:
+            if isinstance(drop, float) and np.isnan(drop):
+                drop = np.isnan(X)
+            else:
+                drop = X == drop
+
+    start = 0
+    while start < len(X) - window_size - target_size + 1:
```

Comment:
Can we put this on a variable to avoid computing it at every iteration?

```
max_start = len(X) - window_size - target_size + 1
while start < max_start:
```